File: frontend/lib/api/contracts/worklist.ts
import { z } from "zod";

export const worklistItemSchema = z.strictObject({
  id: z.uuid(),
  type: z.enum(["REQUEST_PENDING", "RECORDS_READY", "EMERGENCY_REVIEW"]),
  patient_id: z.uuid(),
  patient_name: z.string().min(1).max(200),
  due_at: z.iso.datetime({ offset: false }),
});
export const worklistSchema = z.strictObject({
  items: z.array(worklistItemSchema).max(100),
  next_cursor: z.string().min(1).max(2048).nullable().default(null),
  correlation_id: z.uuid(),
});
export type WorklistItem = z.infer<typeof worklistItemSchema>;

File: backend/app/schemas/worklist.py
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class StrictWorklistModel(BaseModel):
    model_config = ConfigDict(extra="forbid")


class WorklistItem(StrictWorklistModel):
    id: UUID
    type: Literal["REQUEST_PENDING", "RECORDS_READY", "EMERGENCY_REVIEW"]
    patient_id: UUID
    patient_name: str = Field(min_length=1, max_length=200)
    due_at: str = Field(pattern=r"Z$")


class WorklistCollection(StrictWorklistModel):
    items: list[WorklistItem] = Field(max_length=100)
    next_cursor: str | None = Field(default=None, min_length=1, max_length=2048)
    correlation_id: UUID
